tcms/rpc/api/testexecution.py

```diff
@@ -204,6 +204,11 @@ def update(execution_id, values, **kwargs):
     # if this call updated TE.status then adjust timestamps
     if values.get("status"):
         now = timezone.now()
+
+        if form.status.weight != 0:
+            test_execution.stop_date = now
+        test_execution.form.save()
+
```

Comment:
This is already solved on the next few lines. I've updated the issue description so this code block isn't needed.